datasets/cmu_hinglish_dog/cmu_hinglish_dog.py

```diff
@@ -0,0 +1,190 @@
+}
+"""
+
+_DESCRIPTION = """\
+This is a collection of text conversations in Hinglish (code mixing between Hindi-English) and their corresponding English only versions. Can be used for Translating between the two.
+"""
+
+_HOMEPAGE = "http://festvox.org/cedar/data/notyet/"
+_URL_HINGLISH = "http://festvox.org/cedar/data/notyet/CMUHinglishDoG.zip"
+_URL_ENGLISH = (
+    "https://github.com/festvox/datasets-CMU_DoG/archive/master/Conversations.zip"
+)
+
+
+class CMUHinglishDoG(datasets.GeneratorBasedBuilder):
+    """Load the CMU Hinglish DoG Data for MT"""
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "date": datasets.Value("string"),
+                "history": datasets.Sequence(
+                    {
+                        "docIdx": datasets.Value("int64"),
+                        "text": datasets.Value("string"),
+                        "uid": datasets.Value("string"),
+                        "utcTimestamp": datasets.Value("string"),
+                    }
+                ),
+                "rating": datasets.Value("int64"),
+                "status": datasets.Value("int64"),
+                "uid1LogInTime": datasets.Value("string"),
+                "uid1LogOutTime": datasets.Value("string"),
+                "uid1response": datasets.Sequence(
+                    {
+                        "feedback": datasets.Value("string"),
+                        "response": datasets.Sequence([datasets.Value("int64")]),
+                        "type": datasets.Value("string"),
+                    }
+                ),
+                "uid2response": datasets.Sequence(
+                    {
+                        "feedback": datasets.Value("string"),
+                        "response": datasets.Sequence([datasets.Value("int64")]),
+                        "type": datasets.Value("string"),
+                    }
+                ),
+                "user2_id": datasets.Value("string"),
+                "whoSawDoc": datasets.Sequence([datasets.Value("string")]),
+                "wikiDocumentIdx": datasets.Value("int64"),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """The linking part between Hinglish data and English data is inspired from the implementation in GLUECoS.
+        Refer here for the original script https://github.com/microsoft/GLUECoS/blob/7fdc51653e37a32aee17505c47b7d1da364fa77e/Data/Preprocess_Scripts/preprocess_mt_en_hi.py"""
+
+        eng_path = dl_manager.download_and_extract(_URL_ENGLISH)
+
+        # WikiData file is not required for this
+        if os.path.isdir(os.path.join(eng_path, "datasets-CMU_DoG-master", "WikiData")):
+            shutil.rmtree(os.path.join(eng_path, "datasets-CMU_DoG-master", "WikiData"))
```

Comment:
I'm not sure how deleting a directory is handled for the cache - would it be possible to remove this part so that the cache stays intact ?

Reply:
Was causing an issue in automatic dummy data part, fixed now.